extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java

```diff
@@ -81,4 +103,80 @@ private void addReflectiveHierarchyClass(DotName className) {
     private void addReflectiveClass(boolean methods, boolean fields, String... className) {
         reflectiveClass.produce(new ReflectiveClassBuildItem(methods, fields, className));
     }
+
+    // Generate a ObjectMapperCustomizer bean that registers each serializer / deserializer with ObjectMapper
+    @BuildStep
+    void generateCustomizer(BuildProducer<GeneratedBeanBuildItem> generatedBeans,
+            List<CustomJacksonSerDeserBuildItem> customSerDesers) {
+
+        if (customSerDesers.isEmpty()) {
+            return;
+        }
+
+        ClassOutput classOutput = new ClassOutput() {
+            @Override
+            public void write(String name, byte[] data) {
+                generatedBeans.produce(new GeneratedBeanBuildItem(name, data));
+            }
+        };
+
+        try (ClassCreator classCreator = ClassCreator.builder().classOutput(classOutput)
+                .className("io.quarkus.jackson.customizer.RegisterSerializersAndDeserializersCustomizer")
+                .interfaces(ObjectMapperCustomizer.class.getName())
+                .build()) {
+            classCreator.addAnnotation(Singleton.class);
+
+            try (MethodCreator customize = classCreator.getMethodCreator("customize", void.class, ObjectMapper.class)) {
+                ResultHandle objectMapper = customize.getMethodParam(0);
+
+                for (CustomJacksonSerDeserBuildItem customSerDeser : customSerDesers) {
```

Comment:
I don't think I would have the `Custom` in the name.

I wonder if it's such a good idea to couple the serializer and deserializer entirely. Maybe I would have had two different build items.

BTW, if we do that, I would name them `JacksonSerializerBuildItem` and `JacksonDeserializerBuildItem` which is consistent with the Jackson naming.

Reply:
You don't have to set both, you can easily just define one.

The "problem" with Jackson is that the serializers and deserializers need to be part of a module so having one build item makes it easier to define such a module.

I'm of course open to do whatever we think is best.

Reply:
I dropped `Custom` from the name